Controllers/User.go

```diff
@@ -1,109 +1,129 @@
-package Controllers
-import(
+package controller
+
+// package name should start with small letters
+import (
 	"first-api/Models"
 	"fmt"
 	"net/http"
+
 	"github.com/gin-gonic/gin"
 )
 
-
 // get all users
 
-func GetUsers(c *gin.Context){
-	var user []Models.User
-	err := Models.GetAllUsers(&user);
-	if err!= nil{
-		fmt.Println(err.Error())
-		c.AbortWithStatus(http.StatusNotFound)
-	}else{
-		c.JSON(http.StatusOK,user)
-	}
-
+type UserStorer interface {
+	CreateUser(user *model.User) error
+	Validate(user model.User) error
+	GetAllUsers(users *[]model.User) error
+	GetUserByID(user *model.User, id string) error
```

Comment:
This function should be renamed to GetUser and should not be suffixed with ID. 
since let's say later you want to support one more filter named "username, then you have to write getUserByUsername. 

Technically all the filteres passed in query should be validated and injected in query builder while fetching the database.